packages/core/src/components/forms/numericInputUtils.ts

```diff
@@ -30,7 +30,7 @@ function getDecimalSeparator(locale: string) {
 }
 
 export function toLocaleString(num: number, locale: string = "en-US") {
-    return sanitizeNumericInput(num.toLocaleString(locale), locale);
+    return sanitizeNumericInput(num.toLocaleString(locale, { maximumSignificantDigits: 21 }), locale);
```

Comment:
why `21`? according to the [NumberFormat docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#options), this should already be the default. maybe you need to adjust a different option like `roundingMode`?